packages/coachmark/src/CoachmarkPopover.ts

```diff
@@ -0,0 +1,193 @@
+    }
+
+    private async closeOverlay(): Promise<void> {
+        return new Promise(async (res) => {
+            this.open = false;
+            if (this.closeCoachmarkFunction) {
+                this.addEventListener(
+                    'sp-closed',
+                    () => {
+                        res();
+                    },
+                    { once: true }
+                );
+                this.closeCoachmarkFunction();
+                this.closeCoachmarkFunction = undefined;
+            } else {
+                res();
+            }
+            this.dispatchEvent(
+                new Event('sp-coachmark-closed', {
+                    bubbles: true,
+                    cancelable: true,
+                    composed: true,
+                })
+            );
+        });
+    }
+
+    private overlaidContent?: CoachmarkPopoverContent;
+
+    private openOverlay(): Promise<void> {
+        return new Promise(async (res) => {
+            if (!this.content || !this.target || this.closeCoachmarkFunction) {
+                this.open = false;
+                res();
+                return;
+            }
+            this.overlaidContent = this.content;
+            this.overlaidContent.addEventListener('close', (event: Event) => {
+                event.stopPropagation();
+                this.closeOverlay();
+            });
+            this.addEventListener(
+                'sp-opened',
+                () => {
+                    res();
+                },
+                { once: true }
+            );
+            this.closeCoachmarkFunction = await openOverlay(
+                this.target,
+                'custom',
+                this.overlaidContent,
+                this.overlayOptions
+            );
+            this.dispatchEvent(
+                new Event('sp-coachmark-opened', {
+                    bubbles: true,
+                    cancelable: true,
+                    composed: true,
+                })
+            );
+        });
+    }
+
+    protected override async getUpdateComplete(): Promise<boolean> {
+        const complete = (await super.getUpdateComplete()) as boolean;
+        await this.openStatePromise;
+        return complete;
+    }
+
+    public override disconnectedCallback(): void {
+        this.closeOverlay();
+        super.disconnectedCallback();
+    }
+
+    protected override updated(
+        changedProperties: PropertyValueMap<this>
+    ): void {
+        super.updated(changedProperties);
+
+        if (changedProperties.has('open')) {
+            this.manageOpen();
+        }
+    }
+
+    override render(): TemplateResult {
+        return html`
+            <div id="target">
+                <slot></slot>
+                ${when(
+                    this.open,
+                    () => html`
+                        <div id="coachmark-wrapper">
+                            <slot name="pulse">
+                                <sp-coachmark></sp-coachmark>
```

Comment:
Should we be providing an `sp-coachmark` by default for convenience or should we leave this up to the consumer?